File: app/src/main/java/com/yasin/rxjavalearn/Network/ApiInterface.java
package com.yasin.rxjavalearn.Network;

import com.yasin.rxjavalearn.networkCallWithRx.Photo;
import com.yasin.rxjavalearn.networkCallWithRx.User;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.HEAD;
import retrofit2.http.Url;

public interface ApiInterface {

    @GET("albums/1/photos/")
    Observable<List<Photo>> getPhotos();

    @GET("users/")
    Observable<List<User>> getUsers();

    @HEAD()
    Call<Void> checkIfImageExists(@Url String url);
}

File: app/src/main/java/com/yasin/rxjavalearn/Network/ApiInterface.kt
package com.yasin.rxjavalearn.Network

import com.yasin.rxjavalearn.nestedFlatMap.Post
import com.yasin.rxjavalearn.networkCallWithRx.Photo
import com.yasin.rxjavalearn.networkCallWithRx.User

import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.HEAD
import retrofit2.http.Query
import retrofit2.http.Url

interface ApiInterface {

    @get:GET("albums/1/photos/")
    val photos: Observable<List<Photo>>

    @get:GET("users/")
    val users: Observable<List<User>>

    @GET("posts/")
    fun getPosts(@Query("userId") userID: Int): Observable<List<Post>>

    @HEAD
    fun checkIfImageExists(@Url url: String): Call<Void>
}
